### PR TITLE
Sa/build opt in

### PR DESCRIPTION
**Pull request by chefsalim**
This change adds an "Auto Build" option to the Builder project settings. By default, the auto build option is set to Off.  If auto-build is off, Builder will not automatically kick off builds of the package if one of it's reverse dependencies re-builds, or if it detects a commit change via the Github webhook/bldr.toml path. Essentially, only manual builds via the Web UI or the hab client will be conducted.  

tenor-249913659

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-originsrv/src/migrations/2018-06-29-222627_auto_build_opt_in/up.sql [hunk 1]
```diff
@@ -0,0 +1,69 @@
+ALTER TABLE IF EXISTS origin_projects ADD COLUMN IF NOT EXISTS auto_build bool NOT NULL DEFAULT true;
```
Comment (raskchanky) [inline components/builder-originsrv/src/migrations/2018-06-29-222627_auto_build_opt_in/up.sql]: I thought this was supposed to be off by default?

Comment (chefsalim) [reply components/builder-originsrv/src/migrations/2018-06-29-222627_auto_build_opt_in/up.sql]: Yes that was the original plan.. however in the interest of not breaking any existing build workflows (eg, all core plans auto builds would immediately stop), it seems better to default it to true in the DB, and then have a migration path for turning off unneeded builds.  Note that the UI will default auto build to Off as default, so for new plan connections we will have the more conservative behavior.

**Pull request merged**